src/engine/Events/ExEvent.ts

```diff
@@ -0,0 +1,12 @@
+
+export abstract class Vent<TName extends string> {
```

Comment:
rename file to `Vent.ts`?

Reply:
I see, this is because `Event` is reserved. Vent might not be the best because in pub/sub it has a purpose as a way to push messages? Might be confusing 🤷‍♂️ 

Maybe... `EEvent` or `EventBase` or `ExEvent` explicitly even.

Reply:
Good call out, I'll go with ExEvent